DOC Add password strength indicator to changelog (#767)
silverstripe/developer-docs: en/08_Changelogs/6.1.0.md

Note that if you are using [`silverstripe/hybridsessions`](https://packagist.org/packages/silverstripe/hybridsessions) or [`silverstripe/dynamodb`](https://packagist.org/packages/silverstripe/dynamodb), you can continue to use these with no change, provided you update them to the latest version.

> ### Password strength feedback {#password-strength-feedback}

Reviewer: ```suggestion
### Password strength feedback
```
I've mentioned this a few times - an explicit anchor doesn't need to be added unless it differs from what will be implicitly generated. Most IDEs (or, at the very least, vscode which I use) don't know how to handle this explicit syntax so it breaks links as viewed in IDEs.

Author: Updated

> The [`ConfirmedPasswordField`](api:SilverStripe\Forms\ConfirmedPasswordField) now provides real-time feedback on password as users type. You'll see this in action on the member edit form and when setting a new password during the "Lost password" process.

Reviewer: ```suggestion
The [`ConfirmedPasswordField`](api:SilverStripe\Forms\ConfirmedPasswordField) now provides real-time feedback on password strength as users type. You'll see this in action on the member edit form, when setting a new password during the "lost password" process, and for any usages of `ConfirmedPasswordField` in your own forms if [`requireStrongPassword`](api:SilverStripe\Forms\ConfirmedPasswordField->requireStrongPassword) is toggled on.
```

Author: Updated

> This feature is active only when an [`EntropyPasswordValidator`](api:SilverStripe\Security\Validation\EntropyPasswordValidator) is in use for password validation. If you're using the alternative [`RulesPasswordValidator`](api:SilverStripe\Security\Validation\RulesPasswordValidator) or another password validator, this strength indicator won't appear. The strength itself is assessed by Symfony's [PasswordStrength](https://symfony.com/doc/current/reference/constraints/PasswordStrength.html#available-options) constraint validator.

Reviewer: ```suggestion
In Member edit forms and the "lost password" process, this feature is active only when an [`EntropyPasswordValidator`](api:SilverStripe\Security\Validation\EntropyPasswordValidator) is in use for password validation. If you're using the alternative [`RulesPasswordValidator`](api:SilverStripe\Security\Validation\RulesPasswordValidator) or another password validator, this strength indicator won't appear. The strength itself is assessed by Symfony's [`PasswordStrength`](https://symfony.com/doc/current/reference/constraints/PasswordStrength.html#available-options) validation constraint.
```

Author: Updated

> As part of this work, [`ChangePasswordForm`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordForm) now uses a `ConfirmedPasswordField` instead of separate [`PasswordField`](api:SilverStripe\Forms\PasswordField)s. Consequently, the [`ChangePasswordHandler`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordHandler) that processes form submissions has been updated to expect a different data structure. If your site includes any custom modifications to the change password flow, please be aware of this change.

Reviewer: ```suggestion
As part of this work, [`ChangePasswordForm`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordForm) now uses a `ConfirmedPasswordField` instead of separate [`PasswordField`](api:SilverStripe\Forms\PasswordField) instances. Consequently, the [`ChangePasswordHandler`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordHandler) that processes form submissions has been updated to expect a different data structure. If your site includes any custom modifications to the change password flow, please be aware of this change.
```

Reviewer: Need to mention this for MFA's `ResetAccountForm()` as well since it has extension hooks.

Author: Updated

> If your site does not have the [silverstripe/login-forms](https://github.com/silverstripe/silverstripe-login-forms) module installed, then this functionality will not work for the "Lost password" flow out of the box as the required JavaScript will not be available on the frontend theme. This JavaScript can be found [here](https://github.com/silverstripe/silverstripe-login-forms/blob/6/client/src/js/bundle.js) and can be manually added to your frontend theme.

Reviewer: ```suggestion
If your site does not have the [silverstripe/login-forms](https://github.com/silverstripe/silverstripe-login-forms) module installed, then this functionality will not work for the "Lost password" flow out of the box as the required JavaScript will not be available on the frontend theme. This JavaScript can be found [in the login-forms module](https://github.com/silverstripe/silverstripe-login-forms/blob/6/client/src/js/bundle.js) and can be manually added to your frontend theme.
```
Link to the src instead
Also do they need the CSS as well?

Author: I'd rather not link that, the relevant css is only a few lines and it could easily not be compatible / relevant with someones existing css.

Reviewer: Please update the text as per the suggestion ("here" as a link is not accessibility friendly).
Also does "I'd rather not link that" refer to the `src/` of the JavaScript, or just the CSS? I think we should be linking to the source javascript so people can more easily see what's going on and adapt it if they need to - especially if we're not including CSS and therefore they may need to adapt the class names used to match their theme styling.

Author: > Please update the text as per the suggestion

Done

> "I'd rather not link that"

Specifically the CSS. Linking to the JS is what I've already linked to in this PR

> If your site does not have the [silverstripe/login-forms](https://github.com/silverstripe/silverstripe-login-forms) module installed, then this functionality will not work for the "Lost password" flow out of the box as the required JavaScript will not be available on the frontend theme. This JavaScript can be found [here](https://github.com/silverstripe/silverstripe-login-forms/blob/6/client/src/js/bundle.js) and can be manually added to your frontend theme.

Reviewer: ```suggestion
This functionality will not work in your front-end forms out of the box as the required JavaScript will not be available on the frontend theme. If your site does not have the [silverstripe/login-forms](https://github.com/silverstripe/silverstripe-login-forms) module installed, then this includes the "Lost password" flow.

The JavaScript can be found [here](https://github.com/silverstripe/silverstripe-login-forms/blob/6/client/src/js/bundle.js) and can be manually added to your frontend theme.
```

Author: Updated

### Password strength feedback

The [`ConfirmedPasswordField`](api:SilverStripe\Forms\ConfirmedPasswordField) now provides real-time feedback on password strength as users type. You'll see this in action on the member edit form, when setting a new password during the "lost password" process, and for any usages of `ConfirmedPasswordField` in your own forms if [`requireStrongPassword`](api:SilverStripe\Forms\ConfirmedPasswordField->requireStrongPassword) is toggled on.

In Member edit forms and the "lost password" process, this feature is active only when an [`EntropyPasswordValidator`](api:SilverStripe\Security\Validation\EntropyPasswordValidator) is in use for password validation. If you're using the alternative [`RulesPasswordValidator`](api:SilverStripe\Security\Validation\RulesPasswordValidator) or another password validator, this strength indicator won't appear. The strength itself is assessed by Symfony's [`PasswordStrength`](https://symfony.com/doc/current/reference/constraints/PasswordStrength.html#available-options) validation constraint.

> As part of this work, [`ChangePasswordForm`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordForm) now uses a `ConfirmedPasswordField` instead of separate [`PasswordField`](api:SilverStripe\Forms\PasswordField) instances. Consequently, the [`ChangePasswordHandler`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordHandler) that processes form submissions has been updated to expect a different data structure. If your site includes any custom modifications to the change password flow, please be aware of this change. The same is also true for the reset account Form created and processed in the `silverstripe/mfa` in [`SecurityExtension`](api:SilverStripe\MFA\Extension\AccountReset\SecurityExtension).

Reviewer: ```suggestion
As part of this work, [`ChangePasswordForm`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordForm) now uses a `ConfirmedPasswordField` instead of separate [`PasswordField`](api:SilverStripe\Forms\PasswordField) instances. Consequently, the [`ChangePasswordHandler`](api:SilverStripe\Security\MemberAuthenticator\ChangePasswordHandler) that processes form submissions has been updated to expect a different data structure. If your site includes any custom modifications to the change password flow, please be aware of this change. The same is also true for the reset account form created and processed in the [silverstripe/mfa](https://github.com/silverstripe/silverstripe-mfa) module in [`SecurityExtension`](api:SilverStripe\MFA\Extension\AccountReset\SecurityExtension).
```
Note that I removed the backticks around `silverstripe/mfa` to match how login-forms is referenced below - but adding backticks to both is also ok.

Author: Updated
